### PR TITLE
feat: extend with device checkin

### PR DESCRIPTION
**Pull request by merlin-northern**
Changelog: Title
Ticket: MEN-6380

### CODE REVIEW

**Comment by merlin-northern** [comment]
@kjaskiewiczz I have reached the moment where I have the check in time as I want it, with your PR from deviceauth. I need to add a test for it, but also I need to check why 50% of acceptance tests failed, together with units.

**Comment by merlin-northern** [comment]
> @kjaskiewiczz I have reached the moment where I have the check in time as I want it, with your PR from deviceauth. I need to add a test for it, but also I need to check why 50% of acceptance tests failed, together with units.

all pass now.

**Comment by merlin-northern** [comment]
@mender-test-bot start pipeline --fast please

**Review by kjaskiewiczz — APPROVED** [review]
looks good

File: utils/time.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package utils
+
+import "time"
+
+func TruncateToDay(t time.Time) time.Time {
+	return time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, t.Location())
```
[the remark this answers is not in the document]
Comment (merlin-northern) [reply utils/time.go:20]: something like that, but Truncate as such does not guarantee the zero values of the hour, minute, second fields, it does rounding. I feel safer with explicit zeros here.

**Comment by mzedel** [comment]
this seems to miss the docs changes @merlin-northern

**Comment by merlin-northern** [comment]
> this seems to miss the docs changes @merlin-northern

I will fix this in the current PR to reporting.

**Pull request merged**